AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveQueue.swift

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Combine
+import Foundation
+import AWSPluginsCore
+
+@available(iOS 13.0, *)
+protocol ReconcileAndSaveOperationQueue {
+    func addOperation(_ operation: ReconcileAndLocalSaveOperation, modelName: String)
+
+    func cancelOperations(modelName: String)
+
+    func cancelAllOperations()
+
+    func waitUntilOperationsAreFinished()
+
+    init(_ modelSchemas: [ModelSchema])
+}
+
+enum ReoncileAndSaveQueueEvent {
+    case operationAdded(id: UUID)
+    case operationRemoved(id: UUID)
+    case cancelledOperations(modelName: String)
+}
+
+/// A queue used to enqueue `ReconcileAndLocalSaveOperation`s which perform reconcile and save for incoming mutation
+/// sync events to local storage for all model types.
+///
+/// Internally, a `ReconcileAndSaveQueue` will manage an operation queue with concurrency of 1 to perform serial
+/// operations to ensure events are processed in the order in which they are received. For example, an initial sync
+/// will perform sync queries for models based on its dependency order. The results are then processed serially by this
+/// queue to ensure the events are reconciled in the same dependency order in which they are received by the
+/// sync query response.
+///
+/// Additionally, this queue allows per model type cancellations on the operations that are enqueued by calling
+/// `cancelOperations(modelName)`. This allows per model type clean up, while allowing other model reconcilliations to
+/// continue to operate.
+@available(iOS 13.0, *)
+class ReconcileAndSaveQueue: ReconcileAndSaveOperationQueue {
+
+    private let serialQueue = DispatchQueue(label: "com.amazonaws.ReconcileAndSaveQueue.serialQueue",
+                                                 target: DispatchQueue.global())
+    private let reconcileAndSaveQueue: OperationQueue
+    private var modelReconcileAndSaveOperations = [String: [UUID: ReconcileAndLocalSaveOperation]]()
```

Comment:
nit: initialize all members in the same place--either everything at the declaration, or everything in the initializer--so it's easier to keep track of.